overlord/devicestate/firstboot.go

```diff
@@ -264,7 +264,21 @@ func (m *DeviceManager) populateStateFromSeedImpl(tm timings.Measurer) ([]*state
 	}
 
 	chainSorted := func(infos []*snap.Info, infoToTs map[*snap.Info]*state.TaskSet) {
-		sort.Stable(snap.ByType(infos))
+		// This is the order in which snaps will be installed in the
+		// system. We want the boot base to be installed before the
+		// kernel so any existing kernel hook can execute with the boot
+		// base as rootfs.
+		effectiveType := func(info *snap.Info) snap.Type {
+			typ := info.Type()
+			if info.RealName == model.Base() {
+				typ = snap.TypeOS
```

Comment:
my slight preference would be to introduce snap.InternalTypeBootBase and order it at 11 perhaps, instead of reusing TypeOS as TypeOS would one day might finally go away

Reply:
Changed now